lib/active_storage/service/postgresql_service.rb

```diff
@@ -148,10 +148,15 @@ def url_helpers
 
     def url_options
       if ActiveStorage::Current.respond_to?(:url_options)
-        ActiveStorage::Current.url_options
-      else
-        { host: ActiveStorage::Current.host }
+        url_opts = ActiveStorage::Current.url_options
+        return url_opts if url_opts.is_a?(Hash)
       end
+
+      return {
+        protocol: Rails.application.config.app_url_protocol,
+        host: Rails.application.config.app_url_host,
+        port: Rails.application.config.app_url_port,
+      }
```

Comment:
fixes this: https://github.com/AllYourBot/hostedgpt/pull/398#issuecomment-2168135853

the problem is that the `ActiveStorage::Current.url_options` are set only during a request lifecycle (with ActiveStorage::SetCurrent) and workers don't run within the context of a request. This fix also allows you to get the url from within the rails console.

see: https://github.com/rails/rails/blob/main/activestorage/app/controllers/concerns/active_storage/set_current.rb
and: https://github.com/rails/rails/issues/40855#issuecomment-2350744657
and: https://github.com/rails/rails/pull/42520

Reply:
on dang, nice debugging on this. i couldn’t figure out what the difference was within the worker